cmd/kubent/main_test.go

```diff
@@ -85,3 +88,53 @@ func TestStoreCollectorError(t *testing.T) {
 		t.Errorf("Failed to ignore collector with error")
 	}
 }
+
+func TestMainExitCodes(t *testing.T) {
+	testCases := []struct {
+		name     string
+		args     []string // file list
+		expected int      // number of manifests
+	}{
+		{"success", []string{"-c=false", "--helm2=false", "--helm3=false"}, 0},
+		{"error", []string{"-c=not-boolean"}, 2},
+		{"successFound", []string{"-c=false", "--helm2=false", "--helm3=false", "-f=../../fixtures/deployment-v1beta1.yaml"}, 0},
+		{"exitErrorFlagNone", []string{"-c=false", "--helm2=false", "--helm3=false", "-e"}, 0},
+		{"exitErrorFlagFound", []string{"-c=false", "--helm2=false", "--helm3=false", "-e", "-f=../../fixtures/deployment-v1beta1.yaml"}, 200},
+	}
+
+	if os.Getenv("TEST_EXIT_CODE") == "1" {
+		tc, err := strconv.Atoi(os.Getenv("TEST_CASE"))
+		if err != nil {
+			t.Errorf("failed to determin the test case num (TEST_CASE env var): %v", err)
+		}
+
+		os.Args = []string{os.Args[0]}
+		os.Args = append(os.Args, testCases[tc].args...)
+		main()
+		return
+	}
+
+	for i, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var ee *exec.ExitError
+
+			cmd := exec.Command(os.Args[0], "-test.run=TestMainExitCodes")
+			cmd.Env = append(os.Environ(), "TEST_EXIT_CODE=1", "TEST_CASE="+strconv.Itoa(i))
+			err := cmd.Run()
+
+			if tc.expected == 0 && err != nil {
+				t.Errorf("expected to succeed with exit code %d, failed with %v", tc.expected, err)
+			} else if tc.expected != 0 {
```

Comment:
Break up the nested logic here.

Reply:
@david-doit-intl  Pls. have a look, I *think* it's now a bit more readable by using `t.Failf` and un-nesting the logic.